src/hooks/satellites.ts

```diff
@@ -39,10 +45,56 @@ export const useSatellites = create(
             (satellite) => satellite.id !== id
           ),
         })),
+      hideSatellite: (id) =>
+        set((state) => ({
+          ...state,
+          satellites: state.satellites.map((satellite) =>
+            satellite.id === id
+              ? {
+                  ...satellite,
+                  visibility: false,
+                }
+              : satellite
+          ),
+        })),
+      showSatellite: (id) =>
+        set((state) => ({
+          ...state,
+          satellites: state.satellites.map((satellite) =>
+            satellite.id === id
+              ? {
+                  ...satellite,
+                  visibility: true,
+                }
+              : satellite
+          ),
+        })),
+      hideSatellitePath: (id) =>
+        set((state) => ({
+          ...state,
+          satellites: state.satellites.map((satellite) =>
+            satellite.id === id
+              ? {
+                  ...satellite,
+                  path: false,
+                }
+              : satellite
+          ),
+      })),
+      showSatellitePath: (id) =>
+        set((state) => ({
+          ...state,
+          satellites: state.satellites.map((satellite) =>
+            satellite.id === id
+              ? {
+                  ...satellite,
+                  path: true,
+                }
+              : satellite
+          ),
+      })),
+
```

Comment:
This code could be simplified by defining one toggle function instead of two show and hide functions for each option.